sdl_android/src/main/java/com/smartdevicelink/api/screen/TextAndGraphicManager.java

```diff
@@ -0,0 +1,893 @@
+
+			uploadImages(new CompletionListener() {
+				@Override
+				public void onComplete(boolean success) {
+					if (!success){
+						Log.e(TAG, "Error uploading image");
+						inProgressUpdate = extractTextFromShow(inProgressUpdate);
+						sendShow();
+					}
+					// Check if queued image update still matches our images (there could have been a new Show in the meantime)
+					// and send a new update if it does. Since the images will already be on the head unit, the whole show will be sent
+					if ((thisUpdate.getGraphic() != null && queuedImageUpdate.getGraphic() != null) && thisUpdate.getGraphic().equals(queuedImageUpdate.getGraphic()) ||
+							(thisUpdate.getSecondaryGraphic() != null && queuedImageUpdate.getSecondaryGraphic() != null) && thisUpdate.getSecondaryGraphic().equals(queuedImageUpdate.getSecondaryGraphic())){
+						Log.v(TAG, "Queued image update matches the images we need, sending update");
+						sendShow();
+					} else {
+						Log.v(TAG, "Queued image update does not match the images we need, skipping update");
+					}
+				}
+			});
+			queuedImageUpdate = fullShow;
+		}
+	}
+
+	private void sendShow(){
+		inProgressUpdate.setOnRPCResponseListener(new OnRPCResponseListener() {
+			@Override
+			public void onResponse(int correlationId, RPCResponse response) {
+				if (response.getSuccess()){
+					Log.v(TAG, "Show Successful");
+					updateCurrentScreenDataFromShow(inProgressUpdate);
+				}
+
+				inProgressUpdate = null;
+				if (inProgressListener != null){
+					inProgressListener.onComplete(true);
+					inProgressListener = null;
+				}
+
+				if (hasQueuedUpdate){
+					Log.v(TAG, "Queued update exists, sending another update");
+					sdl_update(queuedUpdateListener);
+					queuedUpdateListener = null;
+					hasQueuedUpdate = false;
+				}
+			}
+		});
+
+		this.softButtonManager.setCurrentMainField1(inProgressUpdate.getMainField1());
+		internalInterface.sendRPCRequest(inProgressUpdate);
+	}
+
+	// Images
+
+	private void uploadImages(final CompletionListener listener) {
+
+		List<SdlArtwork> artworksToUpload = new ArrayList<>();
+
+		// add primary image
+		if (shouldUpdatePrimaryImage()){
+			artworksToUpload.add(primaryGraphic);
+		}
+
+		// add secondary image
+		if (shouldUpdateSecondaryImage()){
+			artworksToUpload.add(secondaryGraphic);
+		}
+		Log.i(TAG, "Artworks to upload: "+ artworksToUpload.toString());
+		// use file manager to upload art
+		fileManager.uploadArtworks(artworksToUpload, new MultipleFileCompletionListener() {
+			@Override
+			public void onComplete(Map<String, String> errors) {
+				if (errors != null) {
+					Log.e(TAG, "Error Uploading Artworks. Error: " + errors.toString());
+					listener.onComplete(false);
+				}else{
+					Log.v(TAG, "Successfully uploaded Artworks");
+					listener.onComplete(true);
+				}
+			}
+		});
+	}
+
+	private Show assembleShowImages(Show show){
+
+		if (!shouldUpdatePrimaryImage() && ! shouldUpdateSecondaryImage()){
```

Comment:
Remove unnecessary if statement. The next two if statements will take care of adding the image to the show or not and if not the original `show` will be returned